plugins/base/src/main/kotlin/resolvers/local/DokkaLocationProvider.kt

```diff
@@ -89,24 +89,34 @@ open class DokkaLocationProvider(
 
     private fun getLocalLocation(driWithSourceSets: DRIWithSourceSets, context: PageNode?): String? {
         val (dri, originalSourceSet) = driWithSourceSets
-        val allSourceSets =
+        val allSourceSets: List<Set<DisplaySourceSet>> =
             listOf(originalSourceSet) + originalSourceSet.let { oss ->
                 dokkaContext.configuration.sourceSets.filter { it.sourceSetID in oss.sourceSetIDs }
                     .flatMap { it.dependentSourceSets }
                     .mapNotNull { ssid ->
                         dokkaContext.configuration.sourceSets.find { it.sourceSetID == ssid }?.toDisplaySourceSet()
+                    }.map {
+                        setOf(it)
```

Comment:
Removing `map { setOf(it) }` does not break compilation since `DisplaySourceSet` is a set in itself - can you please leave a comment for why it exists?